crates/cast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,158 @@
+use anyhow::{anyhow, ensure, Context, Ok, Result};
+use camino::Utf8PathBuf;
+use std::fs;
+
+use cast::helpers::constants::SCRIPTS_DIR;
+use cast::helpers::response_structs::ScriptInitResponse;
+use cast::helpers::scarb_utils::get_cairo_version;
+use clap::Args;
+use indoc::{formatdoc, indoc};
+use scarb_metadata::ScarbCommand;
+
+#[derive(Args, Debug)]
+pub struct Init {
+    /// Name of a script to create
+    pub script_name: String,
+}
+
+pub fn init(init_args: &Init) -> Result<ScriptInitResponse> {
+    let script_root_dir_path = get_script_root_dir_path(&init_args.script_name)?;
+
+    ensure!(
+        !script_root_dir_path.exists(),
+        "Script destination `{script_root_dir_path}` already exists"
+    );
+
+    init_scarb_project(&init_args.script_name, &script_root_dir_path)?;
+
+    let modify_files_result = add_dependencies(&script_root_dir_path)
+        .and_then(|()| modify_files_in_src_dir(&init_args.script_name, &script_root_dir_path));
+
+    match modify_files_result {
+        Result::Ok(()) => Ok(ScriptInitResponse {
+            status: format!("Successfully initialized `{}`", init_args.script_name),
+        }),
+        Err(err) => {
+            clean_created_dir_and_files(&script_root_dir_path);
+            Err(err)
+        }
+    }
+}
+
+fn get_script_root_dir_path(script_name: &str) -> Result<Utf8PathBuf> {
+    let current_dir = std::env::current_dir()?;
+
+    let script_root_dir_path = current_dir
+        .file_name()
+        .and_then(|dir_name| dir_name.to_str())
+        .filter(|&dir_name| dir_name == SCRIPTS_DIR)
+        .map_or_else(
+            || current_dir.join(SCRIPTS_DIR).join(script_name),
+            |_| current_dir.join(script_name),
+        );
+
+    Utf8PathBuf::from_path_buf(script_root_dir_path)
+        .map_err(|_| anyhow!("Failed to obtain script root dir path"))
+}
+
+fn init_scarb_project(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    ScarbCommand::new()
+        .args([
+            "new",
+            "--name",
+            &script_name,
+            "--no-vcs",
+            "--quiet",
+            script_root_dir.as_str(),
+        ])
+        .run()
+        .context("Failed to init Scarb project")?;
+
+    Ok(())
+}
+
+fn add_dependencies(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    add_sncast_std_dependency(script_root_dir)
+        .context("Failed to add sncast_std dependency to Scarb.toml")?;
+    add_starknet_dependency(script_root_dir)
+        .context("Failed to add starknet dependency to Scarb.toml")?;
+
+    Ok(())
+}
+
+fn add_sncast_std_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let cast_version = format!("v{}", env!("CARGO_PKG_VERSION"));
```

Comment:
Is this variable available in built package?

Reply:
If I understand correctly this variable is captured at compile time and is embedded into the binary as a static string.

Reply:
Cool, just wanted to make sure. I guess we can just verify by building the binary, and trying to run the init, to be double-sure.

Reply:
I can confirm that it works as expected.